### PR TITLE
feat(experiments): gql resolver for experiments

### LINKED ISSUE #3400: [exp][graphql] experiments connection on dataset with node interface

### PR DESCRIPTION
**Pull request by mikeldking**
resolves #3400 

```graphql
{
  dataset: node(id: "RGF0YXNldDox") {
    ... on Dataset {
      experiments {
        edges {
          node {
            description
          }
        }
      }
    }
  }
}
```

### CODE REVIEW

File: src/phoenix/server/api/queries.py [hunk 1]
```diff
@@ -214,6 +215,20 @@ async def node(self, id: GlobalID, info: Info[Context, None]) -> Node:
...
+                    select(models.Experiment).where(models.Experiment.id == node_id)
+                )
+            if not experiment:
+                raise ValueError(f"Unknown experiment: {id}")
+            return Experiment(
+                id_attr=experiment.id,
+                description=experiment.description,
+                created_at=experiment.created_at,
+                updated_at=experiment.updated_at,
+                metadata=experiment.metadata,
```
Comment (axiomofjoy) [inline src/phoenix/server/api/queries.py:230]:
```suggestion
                metadata=experiment.metadata_,
```

File: app/schema.graphql [hunk 2]
```diff
@@ -247,27 +370,39 @@ type Dimension implements Node {
...
   """
   The segments across both inference sets and returns the counts per segment
   """
   segmentsComparison(primaryTimeRange: TimeRange): Segments!
 }
 
-"""A connection to a list of items."""
+"""
+A connection to a list of items.
+"""
```
Comment (axiomofjoy) [inline app/schema.graphql:396]: This diff is strange. Is it due to a new version of strawberry?

File: src/phoenix/server/api/types/Dataset.py [hunk 3]
```diff
@@ -134,3 +135,34 @@ async def examples(
...
+        query = select(models.Experiment).where(models.Experiment.dataset_id == dataset_id)
+
+        async with info.context.db() as session:
+            experiments = [
+                Experiment(
+                    id_attr=experiment.id,
+                    description=experiment.description,
+                    created_at=experiment.created_at,
+                    updated_at=experiment.updated_at,
+                    metadata=experiment.metadata,
```
Comment (axiomofjoy) [inline src/phoenix/server/api/types/Dataset.py:164]:
```suggestion
                    metadata=experiment.metadata_,
```

**Review by axiomofjoy — APPROVED** [review]

File: app/schema.graphql (hunk 2 above)
Comment (mikeldking) [reply app/schema.graphql:396]: It might be prettier. Let me re-run

**Pull request merged**